### ISSUE TITLE
Add support for 'post type-checking' hooks

### ISSUE THREAD

**Issue by tyrion** [python/mypy#6537]
I am opening this feature request as requested on gitter by @ilevkivskyi .
Currently mypy supports hooks for bases, decorators and metalcass, such as `get_class_decorator_hook`, etc. However those are called after semantic analysis and before type checking. It would be useful to support the same set of hooks to be called after type checking.

**Comment by ilevkivskyi** [comment]
Yes, I think this is reasonable and should be easy to implement. This will also help improving existing plugins where types of class attributes need to be inferred for auto-generated methods.
